packages/oscal-react-library/src/components/OSCALProperties.tsx

```diff
@@ -0,0 +1,179 @@
+        {namespace}
+      </OSCALSystemImplementationTableTitle>
+      <TableContainer sx={{ maxHeight: "20em" }}>
+        <Table aria-label="Components" sx={{ height: "max-content" }}>
+          <StyledTableHead>
+            <TableRow>
+              <StyledHeaderTableCell>Name</StyledHeaderTableCell>
+              <StyledHeaderTableCell>Class</StyledHeaderTableCell>
+              <StyledHeaderTableCell>Value</StyledHeaderTableCell>
+            </TableRow>
+          </StyledTableHead>
+          <TableBody>
+            {properties
+              ?.filter((property: any) => namespaceOf(property?.ns) === namespace)
+              .map((property: any) => (
+                <OSCALProperty property={property} />
+              ))}
+          </TableBody>
+        </Table>
+      </TableContainer>
+    </>
+  );
+};
+
+interface OSCALPropertiesDialogProps {
+  /**
+   * Contains properties elements
+   */
+  properties?: Property[];
+  /**
+   * Title of the element the properties resides under
+   */
+  title: string;
+}
+
+export const OSCALPropertiesDialog = (props: OSCALPropertiesDialogProps) => {
+  const { properties, title } = props;
+
+  if (!properties) {
+    return null;
+  }
+  const namespaceToProps: Record<string, Property[]> = {};
+  for (const prop of properties) {
+    namespaceToProps[namespaceOf(prop.ns)] ??= [];
+    namespaceToProps[namespaceOf(prop.ns)].push(prop);
+  }
+  const { [NIST_DEFAULT_NAMESPACE]: nist, ...thirdParties } = namespaceToProps;
+
+  const [open, setOpen] = React.useState(false);
+  const handleOpen = () => {
+    setOpen(true);
+  };
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  return (
+    <>
+      <OSCALPropertiesButton
+        size="small"
+        variant="outlined"
+        onClick={handleOpen}
+        aria-label={`${title} properties button`}
+        startIcon={<HomeIcon />}
+      >
+        Properties
+      </OSCALPropertiesButton>
+      <Dialog
+        open={open}
+        onClose={handleClose}
+        scroll="paper"
+        aria-labelledby="scroll-dialog-title"
+        aria-describedby="scroll-dialog-description"
+        maxWidth="md"
+        fullWidth
+        sx={{ maxHeight: "75em" }}
+      >
+        <DialogContent sx={{ maxHeight: "75vh" }} dividers>
+          <DialogTitle id="scroll-dialog-title">{title} Properties</DialogTitle>
+          {/* Handle NIST properties */}
+          <OSCALProperties
+            properties={sortProperties(nist)}
+            namespace={NIST_DEFAULT_NAMESPACE}
+            key={NIST_DEFAULT_NAMESPACE}
+          />
+          {
+            /* Handle 3rd party properties */
+            Object.entries(thirdParties)
+              .sort(([key1, props1], [key2, props2]) => key1.localeCompare(key2))
+              .map(([key, props]) => (
+                <OSCALProperties properties={sortProperties(props)} namespace={key} key={key} />
```

Comment:
If it's going to be broken into a different function I'd rather see this as

```ts

function byName(a: Property, b: Property) {
  return a.name.localeCompare(b.name);
}

<OSCALProperties properties={props.sort(byName)} namespace={key} key={key} />
```

This way it's more scriptive _how_ they're being sorted.